src/htm/regions/ScalarSensor.cpp

```diff
@@ -32,19 +32,19 @@ namespace htm {
 
 ScalarSensor::ScalarSensor(const ValueMap &params, Region *region)
     : RegionImpl(region) {
-  params_.size = params.getScalarT<UInt32>("n");
-  params_.activeBits = params.getScalarT<UInt32>("w");
-  params_.resolution = params.getScalarT<Real64>("resolution");
-  params_.radius = params.getScalarT<Real64>("radius");
-  params_.minimum = params.getScalarT<Real64>("minValue");
-  params_.maximum = params.getScalarT<Real64>("maxValue");
-  params_.periodic = params.getScalarT<bool>("periodic");
-  params_.clipInput = params.getScalarT<bool>("clipInput");
+  params_.size = params.getScalarT<UInt32>("n", 0);
+  params_.activeBits = params.getScalarT<UInt32>("w", 0);
+  params_.resolution = params.getScalarT<Real64>("resolution", 0.0);
```

Comment:
these 3 values of "0" are invalid combination, is it what is expected?

Reply:
> these 3 values of "0" are invalid combination, is it what is expected?

Again, taking the defaults from the Spec.  In this case the algorithm also defaults to all 0's.

Reply:
> In this case the algorithm also defaults to all 0's. 

strange, I think all-0 should throw an assert...

Reply:
Well, look at it in another way... You want to insure that the user **Always** specifies at least one of the combinations. If the algorithm still gets all 0's then it should throw an exception.  So taking all defaults is not allowed.